src/main/java/sorald/processor/SerialVersionUidProcessor.java

```diff
@@ -0,0 +1,191 @@
+                    // Here is the special case where the field exisits but it's type is wrong.
+                    // TODO: handle this case after discussion.
+                    return false;
+                }
+                return !hasCorrectModifier(field);
+            } else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if a given class ia a expection for having an ID. SonarQube states 4 expections
+     *
+     * <ul>
+     *   <li>Swing and AWT classes (GUI classes)
+     *   <li>abstract classes
+     *   <li>Throwable and its subclasses (Exceptions and Errors)
+     *   <li>classes marked with @SuppressWarnings("serial")
+     * </ul>
+     *
+     * @param candidate
+     * @return true if the class should be ignored
+     */
+    private boolean isException(CtClass<?> candidate) {
+
+        return candidate.isAbstract()
+                || extendsThrowable(candidate)
+                || supressesWarning(candidate)
+                || isGuiClass(candidate);
+    }
+
+    /**
+     * Tests if a given class is a gui class. A gui class for sonarqube is a class having a swing or
+     * awt class/interface as supertype. See
+     * https://github.com/SonarSource/sonar-java/blob/master/java-checks/src/main/java/org/sonar/java/checks/serialization/SerialVersionUidCheck.java#L66
+     * for their code
+     *
+     * @param candidate
+     * @return true if gui class false otherwise
+     */
+    private boolean isGuiClass(CtClass<?> candidate) {
+        Set<CtTypeReference<?>> superTypes = new HashSet<>();
+        superTypes.add(candidate.getReference());
+        CtType<?> currentClass = candidate;
+        while (currentClass != null) {
+            superTypes.addAll(currentClass.getSuperInterfaces());
+            superTypes.add(currentClass.getSuperclass());
+            if (currentClass.getSuperclass() != null) {
+                currentClass = currentClass.getSuperclass().getTypeDeclaration();
+            } else {
+                currentClass = null;
+            }
+        }
+        superTypes.removeIf(Objects::isNull);
+        for (CtTypeReference<?> ctTypeReference : superTypes) {
+            if (ctTypeReference.getQualifiedName().startsWith("java.awt")
+                    || ctTypeReference.getQualifiedName().startsWith("javax.swing")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * checks if a given class has a @SupressWarnings("serial") annotation
+     *
+     * @param candidate
+     * @return true if present false otherwise
+     */
+    private boolean supressesWarning(CtClass<?> candidate) {
+        SuppressWarnings annotation = candidate.getAnnotation(SuppressWarnings.class);
+        if (annotation != null
+                && Arrays.stream(annotation.value()).anyMatch(v -> v.equals("serial"))) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
I know the docs say that Sonar ignores classes with this suppression, but it does not appear to be the case. I've checked the implementation of the check, and I've tried it in practice: `@SuppressWarnings("serial")` does nothing. So, remove this method.